daffodil-udf/src/test/scala/org/apache/daffodil/udf/TestUdfs.scala

```
@@ -0,0 +1,93 @@
+    } catch {
+      case e: Exception => e.printStackTrace()
+    }
+  }
+
+  @Test def test_Warnings() {
+    try {
+      val udfs = new UDFunctionService(null)
+      val errList = udfs.getErrors
+      val warningsList = udfs.getWarnings.asScala
+      assertEquals(0, errList.size())
+
+      // tests for missing or empty functionClasses
+      assertTrue(warningsList.exists(_.contains(
+        "Provider ignored: [org.badudfs.functionclasses1.StringFunctions.StringFunctionsProvider]." +
+          "\nNo Function Classes found.")))
+
+      // tests for missing annotations
+      assertTrue(warningsList.exists(_.contains(
+        "Provider ignored: [org.badudfs.annotations.StringFunctions.StringFunctionsProvider]." +
+          "\nAnnotations missing for FunctionClass(es):")))
+
+      // tests for empty annotations: missing both
+      assertTrue(warningsList.exists(_.contains(
+        "FunctionClass ignored: [org.badudfs.annotations.StringFunctions.Compare]. " +
+          "From Provider [org.badudfs.annotations.StringFunctions.StringFunctionsProvider]." +
+          "\nAnnotation namespace field is empty or invalid." +
+          "\nAnnotation name field is empty or invalid.")))
+
+      // tests for empty annotations; missing name
+      assertTrue(warningsList.exists(_.contains(
+        "FunctionClass ignored: [org.badudfs.annotations.StringFunctions.Compare]. " +
+          "From Provider [org.badudfs.annotations.StringFunctions.StringFunctionsProvider]." +
+          "\nAnnotation name field is empty or invalid.")))
+
+      // tests non serializable function classes
+      assertTrue(warningsList.exists(_.contains(
+        "Provider ignored: [org.badudfs.nonserializable.StringFunctions.StringFunctionsProvider]. " +
+          "\nFunctionClass(es) must implement java.io.Serializable:" +
+          "\n[org.badudfs.nonserializable.StringFunctions.FuncA]")))
+
+    } catch {
+      case e: Exception => e.printStackTrace()
```

Remove try catch. It masks a failure mode.